api/main.go

```diff
@@ -715,207 +887,256 @@ func (serv *MetadataServer) CreateSourceVariant(ctx context.Context, source *pb.
 	return serv.meta.RequestScheduleChange(ctx, req)
 }
 
-func (serv *MetadataServer) CreateFeatureVariant(ctx context.Context, feature *pb.FeatureVariant) (*pb.Empty, error) {
-	serv.Logger.Infow("Creating Feature Variant", "name", feature.Name, "variant", feature.Variant)
-	return serv.meta.CreateFeatureVariant(ctx, feature)
+func (serv *MetadataServer) CreateFeatureVariant(ctx context.Context, featureRequest *pb.FeatureVariantRequest) (*pb.Empty, error) {
+	requestID, ctx, logger := serv.Logger.InitializeRequestID(ctx)
+	logger = logger.WithResource("feature_variant", featureRequest.FeatureVariant.Name, featureRequest.FeatureVariant.Variant).WithProvider("", featureRequest.FeatureVariant.Provider)
+	logger.Infow("Creating Feature Variant")
+	featureRequest.RequestId = requestID
+
+	return serv.meta.CreateFeatureVariant(ctx, featureRequest)
 }
 
-func (serv *MetadataServer) CreateLabelVariant(ctx context.Context, label *pb.LabelVariant) (*pb.Empty, error) {
-	serv.Logger.Infow("Creating Label Variant", "name", label.Name, "variant", label.Variant)
+func (serv *MetadataServer) CreateLabelVariant(ctx context.Context, labelRequest *pb.LabelVariantRequest) (*pb.Empty, error) {
+	requestID, ctx, logger := serv.Logger.InitializeRequestID(ctx)
+	logger = logger.WithResource("label_variant", labelRequest.LabelVariant.Name, labelRequest.LabelVariant.Variant).WithProvider("", labelRequest.LabelVariant.Provider)
+	label := labelRequest.LabelVariant
+	logger.Infow("Creating Label Variant")
+	labelRequest.RequestId = requestID
+
 	protoSource := label.Source
-	serv.Logger.Debugw("Finding label source", "name", protoSource.Name, "variant", protoSource.Variant)
+	logger.Debugw("Finding label source", "name", protoSource.Name, "variant", protoSource.Variant)
 	source, err := serv.client.GetSourceVariant(ctx, metadata.NameVariant{Name: protoSource.Name, Variant: protoSource.Variant})
 	if err != nil {
 		serv.Logger.Errorw("Could not create label source variant", "error", err)
 		return nil, err
 	}
 	label.Provider = source.Provider()
-	resp, err := serv.meta.CreateLabelVariant(ctx, label)
+	resp, err := serv.meta.CreateLabelVariant(ctx, labelRequest)
 	serv.Logger.Debugw("Created label variant", "response", resp)
 	if err != nil {
 		serv.Logger.Errorw("Could not create label variant", "response", resp, "error", err)
 	}
 	return resp, err
 }
 
-func (serv *MetadataServer) CreateTrainingSetVariant(ctx context.Context, train *pb.TrainingSetVariant) (*pb.Empty, error) {
-	serv.Logger.Infow("Creating Training Set Variant", "name", train.Name, "variant", train.Variant)
+func (serv *MetadataServer) CreateTrainingSetVariant(ctx context.Context, trainRequest *pb.TrainingSetVariantRequest) (*pb.Empty, error) {
+	requestID, ctx, logger := serv.Logger.InitializeRequestID(ctx)
+	logger = logger.WithResource("training_set_variant", trainRequest.TrainingSetVariant.Name, trainRequest.TrainingSetVariant.Variant).WithProvider("", trainRequest.TrainingSetVariant.Provider)
```

Comment:
Lots of these hardcoded empty strings, maybe make a const in logger called:
SkipProviderType which is equal to an empty string